portal_tables/sync_publications.py

```diff
@@ -34,81 +37,109 @@ def add_missing_info(pubs, grants):
     pubs.loc[:, 'Link'] = [
         "".join(["[PMID:", str(pmid), "](", url, ")"])
         for pmid, url
-        in zip(pubs.pubmedId, pubs.pubmedUrl)
+        in zip(pubs['Pubmed Id'], pubs['Pubmed Url'])
     ]
-    pubs['grantName'] = ""
-    for i, row in pubs.iterrows():
-        grant_names = []
-        for g in row['publicationGrantNumber']:
-            grant_names.append(
-                grants[grants.grantNumber == g]['grantName'].values[0])
-        pubs.at[i, 'grantName'] = grant_names
+    
+    pattern = re.compile("(\')([\s\w/-]+)(\')")
+    
+    for col in new_cols: 
+        pubs[col] = ""
+        for row in pubs.itertuples():
+            i = row[0]
+            n = row[4].split(",")
+            extracted = []
```

Comment:
On the rare occasion there are duplicate themes/consortium between the grants, could we maybe use a set here instead of a list?

Then, below, we can add items to the set with `s.update(new_item)` instead of `l.append(new_item)`.

Reply:
Thank you for bringing this up! The duplicate themes and consortium names between grants is definitely an issue I've seen. I added [line 67](https://github.com/mc2-center/mc2-center-dcc/pull/40/files#diff-3f2f07b16780917689682ca0af91d5420dae7335439d92ae28e201f319cd3532R67) to address the issue of duplicate entries:

`clean_values = ', '.join(list((dict.fromkeys(extracted))))`

Do you think this will hold up, or is the set method you described is better suited here?

Reply:
Oh!  I actually didn't explore that line of code, so let me try real quick.  If it works, then let's ignore my set suggestion.